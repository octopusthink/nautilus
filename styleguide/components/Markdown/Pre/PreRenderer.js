import React from 'react';
import PropTypes from 'prop-types';
import cx from 'clsx';
import Styled from 'rsg-components/Styled';
// import prismTheme from '../../../styles/prismTheme';

const styles = ({ space, color, fontSize, fontFamily, borderRadius }) => ({
  pre: {
    fontFamily: fontFamily.monospace,
    fontSize: fontSize.small,
    lineHeight: 1.5,
    color: color.base,
    whiteSpace: 'pre-wrap',
    wordWrap: 'normal',
    tabSize: 2,
    hyphens: 'none',
    backgroundColor: color.codeBackground,
    padding: [[space[1], space[2]]],
    border: [[1, color.codeBackground, 'solid']],
    borderRadius,
    marginTop: 0,
    marginBottom: space[2],
    // ...prismTheme({ color }),
  },
});

export function PreRenderer({ classes, className, children }) {
  const classNames = cx(className, classes.pre);

  const isHighlighted = className && className.indexOf('lang-') !== -1;
  if (isHighlighted) {
    return (
      <pre
        className={classNames}
        dangerouslySetInnerHTML={{ __html: children }}
      />
    );
  }
  return <pre className={classNames}>{children}</pre>;
}

PreRenderer.propTypes = {
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};

export default Styled(styles)(PreRenderer);
