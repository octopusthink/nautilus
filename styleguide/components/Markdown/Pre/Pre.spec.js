import React from 'react';

import Pre from './index';

describe('Markdown Pre', () => {
	it('should render a pre', () => {
		const actual = render(<Pre>This is pre-formatted text.</Pre>);

		expect(actual).toMatchSnapshot();
	});

	it('should render highlighted code', () => {
		const code = '<button>OK</button>';
		const actual = render(<Pre className="lang-html">{code}</Pre>);

		expect(actual).toMatchSnapshot();
	});
});
