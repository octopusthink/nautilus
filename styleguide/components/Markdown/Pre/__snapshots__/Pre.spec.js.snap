// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Pre should render a pre 1`] = `
<pre
  class="rsg--pre-0"
>
  This is pre-formatted text.
</pre>
`;

exports[`Markdown Pre should render highlighted code 1`] = `
<pre
  class="lang-html rsg--pre-0"
>
  <button>
    OK
  </button>
</pre>
`;
